src/index/Index.h

```diff
@@ -377,14 +380,25 @@ class Index {
 
   // no need for explicit instatiation since this function is private
   template <class MetaData>
-  void createPermutationImpl(const string& fileName, const ExtVec& vec,
-                             size_t c0, size_t c1, size_t c2);
+  std::optional<MetaData> createPermutationImpl(const string& fileName,
+                                                const ExtVec& vec, size_t c0,
+                                                size_t c1, size_t c2);
+  template <class MetaData, class Comparator1, class Comparator2>
+  // _______________________________________________________________________
+  void createPermutationPair(ExtVec* vec,
+                             Permutation::PermutationImpl<Comparator1> p1,
+                             Permutation::PermutationImpl<Comparator2> p2,
+                             bool performUnique = false,
+                             bool createPatternsAfterFirst = false);
+  // _______________________________________________________________________
+  template <class MetaData>
+  void exchangeMultiplicities(MetaData* m1, MetaData* m2);
 
   // wrapper for createPermutation that saves a lot of code duplications
   template <class MetaData, class Comparator>
-  void createPermutation(ExtVec* vec,
-                         Permutation::PermutationImpl<Comparator> permutation,
-                         bool performUnique = false);
+  std::optional<MetaData> createPermutation(
```

Comment:
Update the "method comment". Also I'm still wondering when the best time would be to rename this `MetaData` thing to something like `PermOffsets` or something.

Reply:
I think that `PermOffsets` is a good name. But this name convention appears in so many places that I would suggest to clean this up in a different PR.

Reply:
SGTM